src/Compilers/Core/Portable/ConstantValueSpecialized.cs

```diff
@@ -201,6 +210,19 @@ internal override string GetValueToDisplay()
             {
                 return (_value == null) ? "null" : string.Format("\"{0}\"", _value);
             }
+            public override string ToString(string? format, IFormatProvider? provider)
+            {
+                var stringValue = StringValue;
+                if (format is not null && int.TryParse(format, out var len) &&
+                    (len > 0 && len < stringValue.Length))
+                {
+                    return $@"""{stringValue[..len]}"" ...";
```

Comment:
Should do `len - 3` and remove the ` ` before the `...`, otherwise you could have a string that is 11 characters and what we print is longer than the actual constant value was.